### DISCUSSION TITLE
How can I register a py class method into rust as a callback func and delay exec it when event comes？

### DISCUSSION THREAD

**Discussion by sun-rs** [PyO3/pyo3#3011] [Questions] [answered]
I know that pyo3 colud use call_method0 and call_method1 to call a py function. But if I dont want exec it instantly and want to register the func as callback. How to do that？

**Comment by adamreichold** [comment, marked as answer]
Get a handle on the method itself using `let method: Py<PyAny> = object.getattr("method-name")?.into();` where `object` is a ` &PyCell<MyRustClass>`. You can store `method` where you want and when the time comes call it using `method.call1(py, (...))?;`.

**Reply by sun-rs** [reply]
Is it also work around for object is a pure python class as pyany？

**Reply by adamreichold** [reply]
Yes.
